app/database/operator/server_data_operator/handlers/group.ts

```diff
@@ -41,6 +43,34 @@ const GroupHandler = (superclass: any) => class extends superclass implements Gr
             prepareRecordsOnly,
         });
     };
+
+    /**
+     * handleGroupMemberships: Handler responsible for the Create/Update operations occurring on the GroupMembership table from the 'Server' schema
+     * @param {HandleGroupMembershipArgs} groupMembershipsArgs
+     * @param {GroupMembership[]} groupMembershipsArgs.groupMemberships
+     * @param {boolean} groupMembershipsArgs.prepareRecordsOnly
+     * @throws DataOperatorException
+     * @returns {Promise<GroupMembershipModel[]>}
+     */
+    handleGroupMemberships = async ({groupMemberships, prepareRecordsOnly = true}: HandleGroupMembershipArgs): Promise<GroupMembershipModel[]> => {
+        if (!groupMemberships?.length) {
+            // eslint-disable-next-line no-console
+            console.warn(
+                'An empty or undefined "groupMemberships" array has been passed to the handleGroupMemberships method',
+            );
+            return [];
+        }
+
+        const createOrUpdateRawValues = getUniqueRawsBy({raws: groupMemberships, key: 'id'});
```

Comment:
as we talked before, I'm expecting here to see inserts/updates for only those groups that need it, everything else can be discarded to minimize database changes on groups that haven't changed

Reply:
Yup 👍